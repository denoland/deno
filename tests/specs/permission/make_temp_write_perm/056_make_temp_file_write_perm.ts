Deno.mkdirSync("subdir");

// async file
{
  const path = await Deno.makeTempFile({ dir: `subdir` });
  try {
    if (!path.match(/^subdir[/\\][^/\\]+/)) {
      throw Error("bad " + path);
    }
    console.log("good", path);
  } finally {
    await Deno.remove(path);
  }
}
// sync file
{
  const path = Deno.makeTempFileSync({ dir: `subdir` });
  try {
    if (!path.match(/^subdir[/\\][^/\\]+/)) {
      throw Error("bad " + path);
    }
    console.log("good", path);
  } finally {
    await Deno.remove(path);
  }
}

// async dir
{
  const path = await Deno.makeTempDir({ dir: `subdir` });
  try {
    if (!path.match(/^subdir[/\\][^/\\]+/)) {
      throw Error("bad " + path);
    }
    console.log("good", path);
  } finally {
    await Deno.remove(path);
  }
}

// sync dir
{
  const path = Deno.makeTempDirSync({ dir: `subdir` });
  try {
    if (!path.match(/^subdir[/\\][^/\\]+/)) {
      throw Error("bad " + path);
    }
    console.log("good", path);
  } finally {
    await Deno.remove(path);
  }
}
