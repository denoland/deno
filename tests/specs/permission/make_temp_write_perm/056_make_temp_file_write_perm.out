good [WILDCARD]subdir[WILDCARD]
good [WILDCARD]subdir[WILDCARD]
good [WILDCARD]subdir[WILDCARD]
good [WILDCARD]subdir[WILDCARD]
